### PR TITLE
feat: Add a feature flag for Express Build

### PR DESCRIPTION
**Pull request by mshabarov**
Adds a feature flag for Express Build and disables dev server forcibly if this feature flag is set.

Fixes https://github.com/vaadin/flow/issues/15375

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java [hunk 1]
```diff
@@ -404,4 +411,18 @@ private void checkSendUrlsAsParameters() {
                 DEFAULT_SEND_URLS_AS_PARAMETERS);
     }
 
+    private void checkEnableDevServer(boolean log) {
```
Comment (Artur-) [inline flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java:414]: Do we need special handling for dev server in two classes? Can't we just make it in the parent class?

Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java:414]: Shouldn't `enableDevServer`  always have the value of !featureFlagEnabled for now?

Comment (Artur-) [reply flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java:414]: Overriding the value makes no sense. It is always global, never per servlet

Comment (mshabarov) [reply flow-server/src/main/java/com/vaadin/flow/server/DefaultDeploymentConfiguration.java]: Yes, makes sense. Fixed.

**Pull request merged**